src/QMCDrivers/VMC/VMCBatched.cpp

```diff
@@ -0,0 +1,390 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Bryan Clark, bclark@Princeton.edu, Princeton University
+//                    Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//                    Cynthia Gu, zg1@ornl.gov, Oak Ridge National Laboratory
+//                    Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "QMCDrivers/VMC/VMCBatched.h"
+#include "QMCDrivers/VMC/VMCUpdatePbyP.h"
+#include "QMCDrivers/VMC/VMCUpdateAll.h"
+#include "OhmmsApp/RandomNumberControl.h"
+#include "Message/OpenMP.h"
+#include "Message/CommOperators.h"
+#include "Utilities/RunTimeManager.h"
+#include <qmc_common.h>
+//#define ENABLE_VMC_OMP_MASTER
+#include "ADIOS/ADIOS_profile.h"
+#if !defined(REMOVE_TRACEMANAGER)
+#include "Estimators/TraceManager.h"
+#else
+typedef int TraceManager;
+#endif
+
+namespace qmcplusplus
+{
+/// Constructor.
+VMCBatched::VMCBatched(MCPopulation& pop,
+                       TrialWaveFunction& psi,
+                       QMCHamiltonian& h,
+                       WaveFunctionPool& ppool,
+                       Communicate* comm)
+    : QMCDriverNew(pop, psi, h, ppool, comm), UseDrift("yes")
+{
+  RootName = "vmc";
+  QMCType  = "VMCBatched";
+  qmc_driver_mode.set(QMC_UPDATE_MODE, 1);
+  qmc_driver_mode.set(QMC_WARMUP, 0);
+  m_param.add(UseDrift, "useDrift", "string");
+  m_param.add(UseDrift, "usedrift", "string");
+  m_param.add(UseDrift, "use_drift", "string");
+
+  prevSteps               = nSteps;
+  prevStepsBetweenSamples = nStepsBetweenSamples;
+}
+
+bool VMCBatched::run()
+{
+  //   resetRun();
```

Comment:
Is the commented-out code intended to be used in the future?

Reply:
I'm trying to fix it so as much as possible the changes from VMC --> VMCBatched will possible to follow. So some if it is intended to be there in the future some of it will never compile since changes have already been made in the class hierarchy and API's.